perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_sponsoredMembershipInOrganizationsTest.java

```diff
@@ -0,0 +1,101 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Michal Stava stavamichal@gmail.com
+ */
+public class urn_perun_user_attribute_def_virt_sponsoredMembershipInOrganizationsTest {
+	private final static Logger log = LoggerFactory.getLogger(urn_perun_user_attribute_def_virt_sponsoredMembershipInOrganizationsTest.class);
+
+	private final urn_perun_user_attribute_def_virt_sponsoredMembershipInOrganizations classInstance = new urn_perun_user_attribute_def_virt_sponsoredMembershipInOrganizations();
+	private final AttributeDefinition sponsoredMembershipInOrganizationsAttrDef = classInstance.getAttributeDefinition();
+	private final String organizationIdentifierAttrFriendlyName = "sponsorOrganizationIdentifier";
+	private final String organizationIdentifierAttrName = AttributesManager.NS_GROUP_ATTR_DEF + ":" + organizationIdentifierAttrFriendlyName;
+
+	private final User user = new User(1, "Joe", "Doe", "W.", "", "");
+	private final Member member1 = new Member(1, 1, 1, Status.VALID);
+	private final Member member2 = new Member(2, 1, 2, Status.VALID);
+	private final Group group1 = new Group("group1", "group1");
+	private final Group group2 = new Group("group2", "group2");
+
+	private final String value1 = "value1";
+	private final String value2 = "value2";
+
+	private final Attribute groupAttr1 = setUpGroupAttribute(1, organizationIdentifierAttrFriendlyName, String.class.getName(), value1);
+	private final Attribute groupAttr2 = setUpGroupAttribute(1, organizationIdentifierAttrFriendlyName, String.class.getName(), value2);
+
+	private PerunSessionImpl sess;
+
+	@Before
+	public void setUp() throws Exception {
+		sponsoredMembershipInOrganizationsAttrDef.setId(100);
+		group1.setId(1);
+		group2.setId(2);
+
+		//prepare mocks
+		sess = mock(PerunSessionImpl.class);
+		PerunBl perunBl = mock(PerunBl.class);
+		when(sess.getPerunBl()).thenReturn(perunBl);
+
+		AttributesManagerBl am = mock(AttributesManagerBl.class);
+		GroupsManagerBl gm = mock(GroupsManagerBl.class);
+		MembersManagerBl mm = mock(MembersManagerBl.class);
+		when(sess.getPerunBl().getAttributesManagerBl()).thenReturn(am);
+		when(sess.getPerunBl().getGroupsManagerBl()).thenReturn(gm);
+		when(sess.getPerunBl().getMembersManagerBl()).thenReturn(mm);
+
+		when(sess.getPerunBl().getGroupsManagerBl().getGroupsWhereMemberIsActive(sess, member1)).thenReturn(Arrays.asList(group1));
+		when(sess.getPerunBl().getGroupsManagerBl().getGroupsWhereMemberIsActive(sess, member2)).thenReturn(Arrays.asList(group2));
+
+		when(sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, group1, organizationIdentifierAttrName)).thenReturn(groupAttr1);
+		when(sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, group2, organizationIdentifierAttrName)).thenReturn(groupAttr2);
+
+		when(sess.getPerunBl().getMembersManagerBl().getMembersByUserWithStatus(sess, user, Status.VALID)).thenReturn(Arrays.asList(member1, member2));
```

Comment:
NIT: It would be more readable if you would split these lines at the `.thenReturn` call.

Reply:
I like it this way more, but don't have a problem to change that. Done.